test/utils/Permit2Helper.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { IUniswapPermit2 } from "contracts/interfaces/IUniswapPermit2.sol";
+import { TokenCollector } from "contracts/abstracts/TokenCollector.sol";
+import { getEIP712Hash } from "test/utils/Sig.sol";
+import { Test } from "forge-std/Test.sol";
+
+contract Permit2Helper is Test {
```

Comment:
nit: Maybe we could leverage libraries in Permit2 repository to make this simpler in the future PR.

https://github.com/Uniswap/permit2/tree/main/src/libraries